contrib/format-daffodil/src/main/java/org/apache/drill/exec/store/daffodil/DaffodilBatchReader.java

```diff
@@ -64,64 +69,97 @@ public DaffodilBatchReader (DaffodilReaderConfig readerConfig, EasySubScan scan,
     this.validationMode = formatConfig.getValidationMode();
 
     //
-    // FIXME: Next, a MIRACLE occurs.
+    // FIXME: Where is this config file to be found? And, what is its syntax?
```

Comment:
The config file is the `DaffodilFormatConfig` file.  Drill will handle the precedence for you, meaning that if the users defines the variables in a query, that will overwrite the variables stored in the Drill config.

Reply:
That "file" is scala code. You can't mean that's where users put the config info.

Where exactly is the user-created config file found, what naming convention, and what syntax? properties? hocon? json?

Reply:
I'm feeling like we're crossing threads here a bit so let me back up a bit and explain how Drill handles configurations.  When I was talking about configs, I was talking about the params that the format plugin needs.  

## Format Configurations
When you create a format plugin, the first file you likely created was `DaffodilFormatPlugin` which extends the `EasyFormatPlugin` interface.  Inside that generic, you added `DaffodilFormatConfig`. 

```java
public class DaffodilFormatPlugin extends EasyFormatPlugin<DaffodilFormatConfig>
```

By doing this, you've created the format plugin and associated it with a configuration object: `DaffodilFormatConfig`.   We do have the convention of calling these configs: `XXXFormatConfig` or `XXXStorageConfig`, but you could really call it whatever you want as long as that class implements the `FormatPluginConfig` interface.

Let's say that we have a format called `foo`, and we've defined one variable called `bar` in the `FooFormatConfig` class.   Whenever you create a new instance of a file system connection (like HFDS, CP, dfs, etc.) that file system configuration has a list of formats which looks something like this:

```json
 "formats" : {
        "psv" : {
          "type" : "text",
          "extensions" : [ "tbl" ],
          "fieldDelimiter" : "|"
        },
        "csv" : {
          "type" : "text",
          "extensions" : [ "csv" ],
          "fieldDelimiter" : ",",
          "extractHeaders": true
        }
}
```

Using the Drill UI, the user can configure these parameters.  Drill will store the actual values as a json file in zookeeper, however, that process is handled by Drill's internals and isn't something that the format plugin has to manage.  A user should never manually edit these files directly.  They should only do so via Drill's UI, and there is a checksum to enforce that.

As mentioned earlier, the user can override these parameters at query time by using the `table()` functions.  

Drill will handle loading the configuration from this json file for you, and the daffodil format plugin does not need to do anything for that. 

Bottom line is that any parameters you define in your config class, should be available via `plugin.getConfig()`.  Note that there is also a readerConfig object.  This is meant for more complex plugins and i don't think we need to use it for anything in this example.

Does this help?  I hope this answers your questions.

Reply:
No. That didn't help. I'm not looking for user help here. 

A better question is this: where does a junit test put a config file so that the test will read it?
I was trying to test my config file stuff, which was failing in completely inscrutable ways. 
Eventually tracked it down to drills insistence on file extensions. 

My frustration is due to the fact that the entire boilerplate of this plugin configuration stuff could be generated from exactly these lines of specification:
```
name daffodil
schemaURI String null
dataURI String null
rootNamespace String ""
rootName String ""
validationMode Boolean false
```

That's it. The name is "daffodil" and it has these 5 parameters. Spending time debugging just getting to where my code has its 5 parameters has been quite frustrating. 

I have, however, finally got past this and got a first junit test to work end to end. Daffodil constructs drill metadata, Daffodil parses actual data and Drill row is constructed. 

I could use some help to figure out why I can't access things under src/test/resources. I had to use absolute file URLs to get this to work.

Reply:
Let me take a look and see what I can figure out.  I agree that Drill does have too much boilerplate code for things like this.